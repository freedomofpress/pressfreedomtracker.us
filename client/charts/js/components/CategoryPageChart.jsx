import React, { useRef } from 'react'
import { ParentSize } from '@visx/responsive'
import Flashing from '../../../common/js/components/Flashing'
import BarChartYears from "./BarChartYears"
import {
	filterDatasetByFiltersApplied,
	groupByMonthSorted,
	monthIndexes,
	groupByState,
	countIncidentsOutsideUS,
	categoriesColors,
	groupByYearsSorted,
} from '../lib/utilities.js'

import '../../sass/HomepageMainCharts.sass'

export default function CategoryPageChart(props) {
	return (
		<ParentSize>
			{(parent) => <CategoryPageChartWidth {...props} width={parent.width} />}
		</ParentSize>
	)
}

function CategoryPageChartWidth({
	data: dataset,
	width,
	currentDate = new Date(),
	loading = false,
}) {
	const chartWidth = width > 970 ? width / 3 : width
	const chartHeight = width > 970 ? 500 : 480

	const datasetGroupedByYears = groupByYearsSorted(dataset)

	return (
		<Flashing flashing={loading}>
			<div className={'cpChartContainer'} style={{ width: width }}>
				<div className={'cpChart'}>
					<ChartDescription>Showing the number of incidents per year.</ChartDescription>
					{/* <BarChartYears
						width={chartWidth}
						height={chartHeight}
						countYears={countYears}
						selectedYears={selectedYears}
						setSelectedYears={setSelectedYears}
						onClick={onYearClick}
					/> */}
				</div>
			</div>
		</Flashing>
	)
}

function ChartDescription({ children }) {
	return (
		<div
			style={{
				height: '2em',
				padding: 10,
				fontFamily: 'var(--font-base)',
				fontWeight: 400,
				fontSize: 14,
				color: '#7A848E',
				marginBottom: '1rem'
			}}
		>
			{children}
		</div>
	)
}
