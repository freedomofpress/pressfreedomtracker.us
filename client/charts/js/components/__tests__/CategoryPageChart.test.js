import React from 'react'
import * as d3 from 'd3'
import Renderer from 'react-test-renderer'
import CategoryPageChart from '../CategoryPageChart'

test('renders CategoryPageChart with mocked data', () => {
	expect(Renderer.create(
		<CategoryPageChart
			data={[{ categories: ['test'], date: d3.timeParse("%m-%d-%Y")('1-1-2020') }]}
			category={'test'}
			width={500}
			categoryName={'test'}
		/>
	)).toMatchSnapshot();
});
