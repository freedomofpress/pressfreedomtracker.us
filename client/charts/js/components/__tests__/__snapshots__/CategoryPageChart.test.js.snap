// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CategoryPageChart with mocked data 1`] = `
<div
  style={
    {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <div
    className="flashing flashing--off"
  >
    <div
      className="cpChartContainer"
      style={
        {
          "width": 0,
        }
      }
    >
      <div
        className="cpChart"
      >
        <div
          className="chartDescription"
          id="category-page-chart-label"
        >
          Number of 
          test
           incidents per year.
        </div>
      </div>
    </div>
  </div>
</div>
`;
